tests/saveArtifacts.test.ts

```diff
@@ -0,0 +1,76 @@
+import { existsSync, readFileSync } from 'fs';
+import { tmpdir } from 'os';
+import { join } from 'path';
+import { Browser } from 'playwright';
+import * as playwrightVideo from 'playwright-video';
+import { launch } from '../src/launch';
+import { saveArtifacts } from '../src/saveArtifacts';
+import { randomString, waitUntil } from './utils';
+
+describe('saveArtifacts', () => {
+  let browser: Browser;
+
+  beforeAll(async () => {
+    browser = await launch();
+  });
+
+  afterAll(() => browser.close());
+
+  it('saves a video and console logs to specified directory', async () => {
+    const saveDir = join(tmpdir(), randomString());
+
+    const context = await browser.newContext();
+    const page = await context.newPage();
+    const page2 = await context.newPage();
+
+    await saveArtifacts(context, saveDir);
+
+    await page.evaluate(() => console.log('hello'));
+    await page2.evaluate(() => console.info('world'));
+
+    await waitUntil(() => existsSync(join(saveDir, 'logs_0.txt')));
+    await waitUntil(() => existsSync(join(saveDir, 'logs_1.txt')));
+
+    await context.close();
+
+    const testFn = () =>
+      Promise.all([
+        waitUntil(() => existsSync(join(saveDir, 'video_0.mp4'))),
+        waitUntil(() => existsSync(join(saveDir, 'video_1.mp4'))),
+      ]);
+
+    await expect(testFn()).resolves.not.toThrowError();
+    // wait for video capture to finish closing
+    await new Promise(resolve => setTimeout(resolve, 2000));
```

Comment:
@jperl lmk if you have a better suggestion here, if I don't include see open handles warning in Jest